### ISSUE TITLE
Add appxmanifest to the well known file types

### ISSUE THREAD

**Issue by dansiegel** [dotnet/templating#6457]
### Is your feature request related to a problem? Please describe.

Currently a Package.appxmanifest for a WinUI project is not processed automatically as an xml file

### Describe the solution you'd like.

appxmanifest should either be added as a well known extension, or the full file name should be added so that it will follow the xml standards for conditional logic like:

```xml
<!--#if myCondition -->
  <!-- Something here... -->
<!--#endif -->
```

### Additional context

Currently this gets treated as an unknown file type so the condition must be added like:

```xml
//#if myCondition
  <!-- My Content ... -->
//#endif
```

**Comment by YuliiaKovalova** [comment]
Hi @dansiegel,

As a temporary workaround, we can suggest you this special block for `.appxmanifest` processing:
```
"SpecialCustomOperations": {
    "**.appxmanifest": {
      "operations": [
        {
          "type": "conditional",
          "configuration": {
		"endif": [ "#endif", "<!--#endif" ],
		"actionableIf": [ "<!--#if" ],
		"actionableElse": [ "#else", "<!--#else" ],
		"actionableElseif": [ "#elseif", "<!--#elseif", "#elif", "<!--#elif" ],
		"trim": true,
		"wholeLine": true,
		"evaluator": "C++"
          }
        }
      ]
    }
```
Could you specify sdk version that you are working with now?

**Comment by dansiegel** [comment]
@YuliiaKovalova so the AZDO CI build doesn't actually specify any specific SDK version. Locally I'm building with 7.0.202. 

On a related note I might suggest that the `SpecialCustomOperations` be better documented, I've known of it's existence but couldn't remember exactly and could not find it in the Wiki. I would think something in [Conditional processing](https://github.com/dotnet/templating/wiki/Conditional-processing-and-comment-syntax) section would make sense to discuss what it is and how to use it, especially since it's not part of the JSON schema.

**Comment by YuliiaKovalova** [comment]
@dansiegel , we have a section with the example [here](https://github.com/dotnet/templating/blob/main/docs/Reference-for-template.json.md#:~:text=yes-,SpecialCustomOperations,-can%20be%20used).

Great, I hope you are unblocked for now. We will prepare the complex fix.

**Comment by dansiegel** [comment]
yep your first comment was a huge help for sure :)

**Comment by YuliiaKovalova** [comment]
@dansiegel ,

could you tell us if you are ok with this workaround? Because we would like to target code changes like these to .NET 8 already.

**Comment by dansiegel** [comment]
yes, I've got this integrated into the template and it's working as expected.
